Improve grammar in new courses

leila-alderman opened the pull request:
Hi all,

I went through and edited some of the new Ruby courses:
 - Basic Data Types (renamed from "Primitive Data Types")
 - Variables
 - Input and Output
 - Conditional Logic

Let me know if you don't like some of these changes or want to see different things. 

Thanks!
Leila

KevinMulhern commented on new_course/1_programming_fundementals/basic_data_types.md:
```diff
@@ -0,0 +1,323 @@
+    * [Numbers](http://ruby.bastardsbook.com/chapters/numbers/)
+    * [Strings](http://ruby.bastardsbook.com/chapters/strings/)
+
+### Knowledge Check
```
I _really_ like this new knowledge check section, great idea @leila-alderman. Thinking about how it will appear when on the site, do you think this should go above the additional resources section? that would also make it more consistent with the layout of older lessons.

leila-alderman replied:
Glad you like it! I think it makes the beginning of the lessons nice and clean with the much shorter and punchier Learning Objectives. And I won't forget about going back up to quiz myself! 

My thinking in placing this section at the very end was that if you're reading through things and feel like you need more material to understand a concept, you're likely to want that before you quiz yourself. So the quiz is currently placed after all of the content, which also helps to more cleanly group all of the content/resources together. That said, I don't think it's a huge deal whether it comes before or after the Additional Resources. In terms of consistency with the layout of older lessons, though, I've been placing these sections after Additional Resources in the live lessons that I've been editing too (https://www.theodinproject.com/courses/web-development-101/lessons/git-basics). Let me know your thoughts!

KevinMulhern replied:
ah I didn't see that section on Git basics, it seems to fit well at the end of the lesson 😄

leila-alderman replied:
Awesome, I'll keep placing those sections at the end then. Thanks for looking over these changes!

KevinMulhern commented:
Thank you so much for doing this @leila-alderman. You edits are brilliant, these lessons are _much_ cleaner now.

I've left one minor comment, can you let me know your thoughts about it please? 😄

leosoaivan commented on new_course/1_programming_fundementals/conditional_logic.md:
````diff
@@ -83,135 +81,135 @@ end
 ```ruby
-5 == 5 #=> true
-5 == 6 #=> false
+5 == 5 # => true
````
Can we keep this as `#=>` for consistency with other lessons? I believe it's also convention to have it unspaced, but I could be wrong. @KevinMulhern?

leila-alderman replied:
@leosoaivan Thanks for looking over these edits! 

Before I made these changes, I did a Google search to see if I could find this symbol used as a convention elsewhere, but I didn't find anything. Perhaps you can point me to where this is documented. 

I made this change for two reasons:
1. According to [multiple](https://www.digitalocean.com/community/tutorials/how-to-use-comments-in-ruby) [Ruby style guides](https://github.com/rubocop-hq/ruby-style-guide#hash-space), the appropriate convention is to add a space after the comment hash. 
2. Without this space, I felt it could be confusing to new students who might try to read "#=>" as all one symbol, which it's not. By adding the space, it makes it visually clearer that the "#" starts a comment and that the "=>" is just an annotation. 

In terms of consistency with other lessons, I plan to edit all of the lessons, and this change is easy enough to make throughout depending on the consensus here.

leila-alderman replied:
Awesome! Thanks for pointing that out. I love me some sweet, sweet documentation. I'll add another commit to undo those spaces.

leosoaivan commented:
I left a comment. I also made some copyedits that I think would be good to implement. If we pull this in, I can just go over it again.

leila-alderman commented:
@leosoaivan Sorry about the conflicting PR! I've added a commit to incorporate the changes that you wanted to implement. The only change that I didn't make was your suggestion of changing `if; else` to `if..else`. I tried to Google around to find some consensus on the syntax here, but I didn't come across something that seemed standard. If the latter is preferred, I think a full ellipses (`if...else`) is more common. Let me know what you think.

KevinMulhern commented:
👍 LGTM
Once @leosoaivan gives his approval we'll get this merged 😄 Thanks again @leila-alderman

leosoaivan commented:
@leila-alderman I think the full ellipses is the way to go, definitely over `if; else`.

W3Schools and MDN both refer to them as `if...else`, though their documentation is JavaScript.

leila-alderman commented:
I changed all of the `if; else` statements to `if...else` syntax. 

I also added an Assignment section based on the conversation in https://github.com/TheOdinProject/curriculum/pull/9324. I also added in more Additional Resources from the current lesson and added a section on the spaceship operator.
